### PR TITLE
use spring-io platform third party recommendations

### PR DESCRIPTION
**Pull request by i022098**
Signed-off-by: Yoel Lerner <yoel.lerner@hp.com>

### CODE REVIEW

File: engine/data/score-data-api/src/main/java/io/cloudslang/engine/dialects/ScoreDialectResolver.java [hunk 1]
```diff
@@ -37,6 +37,7 @@ protected Dialect resolveDialectInternal(DatabaseMetaData metaData) throws SQLEx
         if ( "MySQL".equals( databaseName ) ) {
 			return new ScoreMySQLDialect();
         }
-        return super.resolveDialectInternal(metaData);
+        DatabaseMetaDataDialectResolutionInfoAdapter databaseMetaDataDialectResolutionInfoAdapter=new DatabaseMetaDataDialectResolutionInfoAdapter(metaData);
+        return resolveDialect(databaseMetaDataDialectResolutionInfoAdapter);
```
Comment (meirwah) [inline engine/data/score-data-api/src/main/java/io/cloudslang/engine/dialects/ScoreDialectResolver.java]: did you validate that the MySql as a DB still works?

File: engine/data/score-data-api/pom.xml [hunk 2]
```diff
@@ -25,17 +25,17 @@
...
-			<groupId>org.hibernate.javax.persistence</groupId>
-			<artifactId>hibernate-jpa-2.0-api</artifactId>
-		</dependency>
-
-		<dependency>
-			<groupId>org.hibernate</groupId>
-			<artifactId>hibernate-core</artifactId>
+		
+		<dependency>
+			<groupId>org.eclipse.persistence</groupId>
```
Comment (meirwah) [inline engine/data/score-data-api/pom.xml]: ????

Comment (i022098) [inline engine/data/score-data-api/pom.xml]: As of Spring IO Platform 1.1, the dependency management for org.hibernate.javax.persistence:hibernate-jpa-2.0-api is deprecated due to the use of a version number in its artifact ID. org.eclipse.persistence:javax.persistence is its preferred replacement.

File: engine/data/score-data-api/src/main/java/io/cloudslang/engine/dialects/ScoreDialectResolver.java (hunk 1 above)
Comment (i022098) [inline engine/data/score-data-api/src/main/java/io/cloudslang/engine/dialects/ScoreDialectResolver.java]: yes i did

**Pull request merged**